tensorboard/plugins/hparams/list_session_groups.py

```diff
@@ -968,3 +972,59 @@ def _build_data_provider_sort_item(col_param):
         hyperparameter_name=col_param.hparam,
         sort_direction=sort_direction,
     )
+
+
+def _get_hparams_to_include(col_params):
+    """Generates the list of hparams to include in the response.
+
+    The determination is based on the `include_in_result` field in ColParam. If
+    a ColParam either has `include_in_result: True` or does not specify the
+    field at all, then it should be included in the result.
+
+    Args:
+      col_params: A collection of `ColParams` protos.
+
+    Returns:
+      A list of names of hyperparameters to include in the response. If None,
+      all hyperparameters will be included.
+    """
+    # If none of the col_params contains `include_in_result` field, all hparams should be included in the response.
+    if all(
+        not col_param.HasField("include_in_result") for col_param in col_params
+    ):
+        return None
```

Comment:
This pattern of calling a "get" method then returning `None` to indicate that all should be included is quite confusing. Can we just return all the params? That should reduce amount of places the code needs to care about this and minimize the number of logical branches.
```suggestion
        return [col_param.hparam for col_param in col_params]
```

I would also be okay with renaming the function, or returning something more meaningful than `None`.

Reply:
Good point! My only concern is if there are a large number of Hparams, then we will be passing all of them to the backend and not sure if it will worsen the performance, I'll see if I could do some testing internally later.

Reply:
@rileyajones I added @bmd3k's helper method `_specifies_include()` (https://github.com/bmd3k/tensorboard/commit/ab75cd1812097f5ca54085e68a34a8cfa346a60b) so that the get method is cleaner but we can still do a quick check to avoid sending all the hparams when no `include` field is specified.

Reply:
The internal tests are not passing, will fix this PR and let you know once it's ready.

Reply:
Actually tests passed after removing other changes outside this PR (cl/563163418).